dog.md

```diff
@@ -0,0 +1,208 @@
+ // -------------------------------------
+
+    syntax DogFile ::= "vow"  Address
+                     | "Hole" Rad
+                     | "hole" String Rad
+                     | "chop" String Wad
+                     | "clip" String Address
+ // -----------------------------------------
+    rule <k> Dog . file vow DOG_VOW => . ... </k>
+         <dog-vow> _ => DOG_VOW </dog-vow>
+
+    rule <k> Dog . file Hole HOLE => . ... </k>
+         <dog-hole> _ => HOLE </dog-hole>
+
+    rule <k> Dog . file hole ILK HOLE => . ... </k>
+         <dog-ilks> ... ILK |-> Ilk( ... hole: (_ => HOLE) ) ... </dog-ilks>
+
+    rule <k> Dog . file chop ILK CHOP => . ... </k>
+         <dog-ilks> ... ILK |-> Ilk( ... chop: (_ => CHOP) ) ... </dog-ilks>
+      requires CHOP >=Wad wad(1)
+
+    rule <k> Dog . file clip ILK CLIP => . ... </k>
+         <dog-ilks> ... ILK |-> Ilk( ... clip: (_ => CLIP) ) ... </dog-ilks>
+         <clip-ilk> CLIP_ILK </clip-ilk>
+      requires ILK ==K CLIP_ILK
+```
+
+Dog Semantics
+-------------
+
+```k
+    syntax DogStep ::= "bark" String Address Address
+ // ------------------------------------------------
+    rule <k> Dog . bark ILK URN KPR
+    =>  #let ROOM = minRad( ( DOG_HOLE -Rad DOG_DIRT), (HOLE -Rad  DIRT) ) #in (
+        #let DART_INITIAL = minWad( ART, ( ( ROOM /Rate RATE) /Wad CHOP) ) #in (
+        #let DART_FINAL = (#if (ART >Wad DART_INITIAL) andBool ( ((ART -Wad DART_INITIAL) *Rate RATE) <Rad DUST ) #then ART #else   DART_INITIAL #fi) #in (
+        #let NON_DUSTY = (#if (ART >Wad DART_INITIAL) andBool ( ((ART -Wad DART_INITIAL) *Rate RATE) <Rad DUST ) #then true #else (DART_INITIAL *Rate RATE) >=Rad DUST  #fi) #in (
+        #let DINK = (INK *Wad DART_FINAL) /Wad ART #in (
+        #let DUE = DART_FINAL *Rate RATE #in (
+        #let TAB = DUE *RadWad2Rad CHOP #in (
+    NON_DUSTY
+    ~> KPR
+    ~> TAB
+    ~> emitBark ILK URN DINK DART_FINAL DUE CLIP 0 ) ) ) ) ) ) )
+    ... </k>
+         <dog-live> DOG_LIVE  </dog-live>
+         <dog-dirt> DOG_DIRT  </dog-dirt>
+         <dog-hole> DOG_HOLE  </dog-hole>
+         <vat-urns> ... { ILK, URN } |-> Urn( INK, ART ) ...                                                </vat-urns>
+         <dog-ilks> ... ILK |-> Ilk( ... clip: CLIP:ClipContract, chop: CHOP, hole: HOLE,  dirt: DIRT ) ... </dog-ilks>
+         <vat-ilks> ... ILK |-> Ilk( ... rate: RATE, spot: SPOT, dust: DUST ) ...                           </vat-ilks>
+      requires DOG_LIVE ==Bool true
+      andBool ( ( SPOT >Ray ray(0) )
+      andBool ( (INK *Rate SPOT ) <Rad (ART *Rate RATE) ) )
+      andBool ( (DOG_HOLE >Rad DOG_DIRT )
+      andBool ( HOLE >Rad DIRT ) )
+
+    rule <k> NON_DUSTY:Bool ~> KPR:Address ~> TAB:Rad ~> emitBark ILK URN DINK DART_FINAL DUE CLIP _
+    => call DOG_VAT . grab ILK URN CLIP DOG_VOW (wad(0) -Wad DINK) (wad(0) -Wad DART_FINAL)
+    ~> call DOG_VOW . fess DUE
+    ~> call CLIP . kick TAB DINK URN KPR
+    ~> emitBark ILK URN DINK DART_FINAL DUE CLIP 0
```

Comment:
Same comment here about last event argument.

Reply:
Okay, given the discussion above I'd expect to see here some non-zero value for the id, even if we cheat and read it from the Clipper's storage. Or maybe I'm still missing something :sweat_smile: .

Reply:
I did not include `id` in the Clippers storage, I believe `id` is not a variable that is maintained in storage in the solidity implementation as well right? I believe it is only used in memory. It is still a 0 value and I explain why on the comment where I tagged you.

Reply:
Coming back to this a few months later--maybe I finally understand. The auction id is going to come from the contents of the k cell prior to continuing through to `emitBark`, because it is the return value of `kick`, right?